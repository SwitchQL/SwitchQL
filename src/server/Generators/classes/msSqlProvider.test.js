import MSSqlProvider from "./msSqlProvider";

describe("MSSqlProvider", () => {
  let provider;

  beforeAll(() => {
    provider = new MSSqlProvider();
  });

  it("Should generate code for selectWithWhere that matches the snapshot", () => {
    const result = provider.selectWithWhere(
      "testTable",
      "testCol",
      "parent.id"
    );
    expect(result).toMatchSnapshot();
  });

  it("Should generate correct sql for selectWithWhere", () => {
    const expected = `SELECT * FROM "testTable" WHERE "testCol" = \${parent.id}`;
    const result = provider.selectWithWhere(
      "testTable",
      "testCol",
      "parent.id"
    );

    expect(result).toContain(expected);
  });

  it("Should generate code for select that matches the snapshot", () => {
    const result = provider.select("testTable");
    expect(result).toMatchSnapshot();
  });

  it("Should generate correct sql for select", () => {
    const expected = `'SELECT * FROM "testTable"'`;
    const result = provider.select("testTable");

    expect(result).toContain(expected);
  });

  it("Should generate code for insert that matches the snapshot", () => {
    const result = provider.insert(
      "testTable",
      "testCol",
      "blah = ${parent.blah}, test = ${parent.test}"
    );
    expect(result).toMatchSnapshot();
  });

  it("Should generate correct sql for insert", () => {
    const expected = `INSERT INTO "testTable" (col1, col2) OUTPUT INSERTED.* VALUES (blah = \${parent.blah}, test = \${parent.test})`;
    const result = provider.insert(
      "testTable",
      "col1, col2",
      "blah = ${parent.blah}, test = ${parent.test}"
    );

    expect(result).toContain(expected);
  });

  it("Should generate code for update that matches the snapshot", () => {
    const result = provider.update("testTable", "idCol");
    expect(result).toMatchSnapshot();
  });

  it("Should generate correct sql for update", () => {
    const expected = `UPDATE "testTable" SET \${parameterized} OUTPUT INSERTED.* WHERE "idCol" = @idCol`;
    const result = provider.update("testTable", "idCol");

    expect(result).toContain(expected);
  });

  it("Should generate code for delete that matches the snapshot", () => {
    const result = provider.delete("testTable", "idCol");
    expect(result).toMatchSnapshot();
  });

  it("Should generate correct sql for delete", () => {
    const expected = `DELETE FROM "testTable" OUTPUT DELETED.* WHERE "idCol" = \${args.idCol}`;
    const result = provider.delete("testTable", "idCol");

    expect(result).toContain(expected);
  });

  it("Should generate code for paramaterize that matches the snapshot", () => {
    const result = provider.parameterize();
    expect(result).toMatchSnapshot();
  });

  it("Should generate code for configureExport that matches the snapshot", () => {
    const result = provider.configureExport();
    expect(result).toMatchSnapshot();
  });
});
