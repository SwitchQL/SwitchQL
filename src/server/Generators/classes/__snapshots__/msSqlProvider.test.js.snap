// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MSSqlProvider Should generate code for configureExport that matches the snapshot 1`] = `
"module.exports = function(connection) { 
              pool = connection;
              return new GraphQLSchema({
                query: RootQuery,
                mutation: Mutation
              });
            }"
`;

exports[`MSSqlProvider Should generate code for delete that matches the snapshot 1`] = `
"return pool.query\`DELETE FROM \\"testTable\\" OUTPUT DELETED.* WHERE \\"idCol\\" = \${args.idCol}\`
          .then(data => {
            return data.recordset[0];
          })
          .catch(err => {
            return ('The error is', err);
          })"
`;

exports[`MSSqlProvider Should generate code for insert that matches the snapshot 1`] = `
"return pool.query\`INSERT INTO \\"testTable\\" (testCol) OUTPUT INSERTED.* VALUES (blah = \${parent.blah}, test = \${parent.test})\`
          .then(data => {
            return data.recordset[0];
          })
          .catch(err => {
            return ('The error is', err);
          })"
`;

exports[`MSSqlProvider Should generate code for paramaterize that matches the snapshot 1`] = `
"        let updateValues = [];
        const req = pool.request();
        for (const prop in rest) {
            req.input(\`\${prop}\`, rest[prop]);
            updateValues.push(\`\${prop} = @\${prop}\`);
        }

        const parameterized = updateValues.join(\\", \\");
"
`;

exports[`MSSqlProvider Should generate code for select that matches the snapshot 1`] = `
"return pool.query('SELECT * FROM \\"testTable\\"')
          .then(data => {
            return data.recordset;
          })
          .catch(err => {
            return ('The error is', err);
          })"
`;

exports[`MSSqlProvider Should generate code for selectWithWhere that matches the snapshot 1`] = `
"return pool.query\`SELECT * FROM \\"testTable\\" WHERE \\"testCol\\" = \${parent.id}\`
          .then(data => {
            return data.recordset[0];
          })
          .catch(err => {
            return ('The error is', err);
          })"
`;

exports[`MSSqlProvider Should generate code for update that matches the snapshot 1`] = `
"        req.input('idCol', idCol);
return req.query(\`UPDATE \\"testTable\\" SET \${parameterized} OUTPUT INSERTED.* WHERE \\"idCol\\" = @idCol\`)
          .then(data => {
            return data.recordset[0];
          })
          .catch(err => {
            return ('The error is', err);
          })"
`;
